src/App.tsx

```diff
@@ -1,54 +1,95 @@
-import React from 'react';
-import './App.scss';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+
 import { peopleFromServer } from './data/people';
 
+import './App.scss';
+// import { Person } from './types/Person';
+
 export const App: React.FC = () => {
+  const [change, setChange] = useState('');
+
   const { name, born, died } = peopleFromServer[0];
+  const [appliedValue, setAppliedValue] = useState('');
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+
+  const handleDebouncedChange = useCallback(debounce((inputValue: string) => {
+    setAppliedValue(inputValue);
+  }, 1000), []);
+
+  const handleInputFocus = () => {
+    setIsDropdownActive(true);
+  };
+
+  const handleInputBlur = () => {
+    setIsDropdownActive(false);
+  };
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const inputValue = event.target.value;
+
+    setChange(inputValue);
+    handleDebouncedChange(inputValue);
+  };
+
+  const filterPerson = useMemo(() => {
+    return peopleFromServer.filter(perso => perso.name
+      .toLowerCase().includes(appliedValue.toLowerCase()));
+  }, [appliedValue]);
+
+  const handleClick = useCallback((persName: string) => {
+    setChange(persName);
+    setIsDropdownActive(false);
+  }, []);
 
   return (
     <main className="section">
       <h1 className="title">
         {`${name} (${born} = ${died})`}
       </h1>
 
-      <div className="dropdown is-active">
+      <div className={`dropdown ${isDropdownActive ? 'is-active' : ''}`}>
```

Comment:
Use [classnames](https://www.npmjs.com/package/classnames) library to add classes conditionally